src/api/python/hidra/utils/utils_general.py

```diff
@@ -371,6 +372,35 @@ def create_sub_dirs(dir_path, subdirs, dirs_not_to_create=()):
         raise OSError
 
 
+class TempFile:
+    def __init__(self, temp_file, target_path):
```

Comment:
I'd suggest docstrings here (or for the class definition)

Reply:
Done